v0/destinations/hs/HSTransform-v1.js

```diff
@@ -0,0 +1,350 @@
+
+//   if (!qualifiedName) {
+//     throw new Error(
+//       "HubSpot qualifiedName is not provided. Aborting custom-object association",
+//       400
+//     );
+//   }
+
+//   if (!objects || !Array.isArray(objects) || objects.length === 0) {
+//     throw new Error(
+//       "HubSpot objects are not provided.  Aborting custom-object association",
+//       400
+//     );
+//   }
+
+//   const endpoint = CRM_ASSOCIATION_V3.replace(
+//     ":fromObjectType",
+//     qualifiedName
+//   ).replace(":toObjectType", "contact");
+
+//   const inputs = [];
+//   objects.forEach(item => {
+//     inputs.push({
+//       from: { id: item.objectId },
+//       to: { id: contactId },
+//       type: `${item.objectType}_to_contact`
+//     });
+//   });
+
+//   // creating response
+//   response = defaultRequestConfig();
+//   response.endpoint = endpoint;
+//   response.headers = {
+//     "Content-Type": "application/json"
+//   };
+//   response.body.JSON = { inputs };
+
+//   // choosing API Type
+//   if (Config.authorizationType === "newPrivateAppApi") {
+//     // Private Apps
+//     response.headers = {
+//       ...response.headers,
+//       Authorization: `Bearer ${Config.accessToken}`
+//     };
+//   } else {
+//     // use legacy API Key
+//     response.params = { hapikey: Config.apiKey };
+//   }
+
+//   return response;
+// };
+
+/**
+ * using legacy API
+ * Ref - https://legacydocs.hubspot.com/docs/methods/enterprise_events/http_api
+ * @param {*} message
+ * @param {*} destination
+ * @param {*} propertyMap
+ * @returns
+ */
+const processLegacyTrack = async (message, destination, propertyMap) => {
+  const { Config } = destination;
+  let parameters = {
+    _a: Config.hubID,
+    _n: message.event,
+    _m: get(message, "properties.revenue") || get(message, "properties.value"),
+    id: getDestinationExternalID(message, "hubspotId")
+  };
+
+  parameters = removeUndefinedAndNullValues(parameters);
+  const userProperties = await getTransformedJSON(
+    message,
+    hsCommonConfigJson,
+    destination,
+    propertyMap
+  );
+
+  const payload = { ...parameters, ...userProperties };
+  const params = removeUndefinedAndNullValues(payload);
+
+  const response = defaultRequestConfig();
+  response.endpoint = TRACK_ENDPOINT;
+  response.method = defaultGetRequestConfig.requestMethod;
+  response.headers = {
+    "Content-Type": "application/json"
+  };
+
+  // choosing API Type
+  if (Config.authorizationType === "newPrivateAppApi") {
+    // eslint-disable-next-line no-underscore-dangle
+    delete params._a;
+    response.headers = {
+      ...response.headers,
+      Authorization: `Bearer ${Config.accessToken}`
+    };
+  }
+  response.params = params;
+
+  return response;
+};
+
+const legacyBatchEvents = destEvents => {
+  const batchedResponseList = [];
+  const trackResponseList = [];
+  const eventsChunk = [];
+  destEvents.forEach(event => {
+    // handler for track call
+    if (event.message.method === "GET") {
```

Comment:
rather than using method for segregating track/identify use some root property like we did for RETL for distinction which is much safe

Reply:
response.messageType = "track" can be used here